### PR TITLE
Support read plugin from config file

### PR DESCRIPTION
**Pull request by RanMaoyi**
For example:

```
plugin {
    libs = [{
        path = "/path/to/plugin_alpha.so"
    }, {
        path = "/path/to/xxx.so"
    }]
}
```

### CODE REVIEW

**Review by OdyWayne — APPROVED** [review]

File: src/supplemental/nanolib/conf_ver2.c [hunk 1]
```diff
@@ -1028,10 +1028,45 @@ conf_bridge_parse_ver2(conf *config, cJSON *jso)
 	return;
 }
 
+#if defined(SUPP_PLUGIN)
+static void
+conf_plugin_parse_ver2(conf *config, cJSON *jso)
+{
+	cJSON *plugin = hocon_get_obj("plugin", jso);
```
Comment (lee-emqx) [inline src/supplemental/nanolib/conf_ver2.c:1035]: Maybe use  hocon_get_obj in this way `cJSON *libs = hocon_get_obj("plugin.libs", jso)` is better.

File: include/nng/supplemental/nanolib/conf.h [hunk 2]
```diff
@@ -310,6 +310,14 @@ struct conf_exchange {
 	conf_exchange_node **nodes;
 };
 
+#if defined(SUPP_PLUGIN)
+typedef struct conf_plugin conf_plugin;
+struct conf_plugin {
+	char **path;
+	size_t path_sz;
+};
+#endif
```
Comment (wanghaEMQ) [inline include/nng/supplemental/nanolib/conf.h:319]: Structure `conf_plugin` should align with the structure of it in config.

for example. the structure of plugin in configuration is 
```
plugin {
    libs = [{
        path = "/path/to/plugin_alpha.so"
    }, {
        path = "/path/to/xxx.so"
    }]
}
```

Then the structure of conf_plugin should be 
```
struct conf_plugin_lib {
    char *path;
}
struct conf_plugin {
    struct conf_plugin_lib **libs;
    size_t len;
}
```

File: src/supplemental/nanolib/conf_ver2.c [hunk 3]
```diff
@@ -1028,10 +1028,45 @@ conf_bridge_parse_ver2(conf *config, cJSON *jso)
...
+	config->plugin.path_sz  = 0;
+
+	libs = hocon_get_obj("libs", plugin);
+
+	cJSON_ArrayForEach(lib, libs) {
+		cJSON *jso_path = cJSON_GetObjectItem(lib, "path");
+		if (jso_path) {
+			if (cJSON_IsString(jso_path)) {
+				if (jso_path->valuestring != NULL) {
+					log_error("jso_path->valuestring: %s", jso_path->valuestring);
```
Comment (wanghaEMQ) [inline src/supplemental/nanolib/conf_ver2.c:1049]: Donot forget to remove the log when PR is already. Maybe changing its log level to info is also fine

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

File: src/supplemental/nanolib/conf_ver2.c (hunk 3 above)
Comment (RanMaoyi) [reply src/supplemental/nanolib/conf_ver2.c]: TKS, removed

File: include/nng/supplemental/nanolib/conf.h (hunk 2 above)
Comment (RanMaoyi) [reply include/nng/supplemental/nanolib/conf.h:325]: Got it, done

File: src/supplemental/nanolib/conf_ver2.c (hunk 1 above)
Comment (RanMaoyi) [reply src/supplemental/nanolib/conf_ver2.c]: Good suggestion, already done

**Review by wanghaEMQ — APPROVED** [review]

**Comment by wanghaEMQ** [comment]
Well done

**Comment by wanghaEMQ** [comment]
How about adding a print_plugin_conf function at your next PR? @RanMaoyi
So when plugin is configured then the content will be printed at `conf.c:1345`.

**Comment by RanMaoyi** [comment]
> How about adding a print_plugin_conf function at your next PR? @RanMaoyi So when plugin is configured then the content will be printed at `conf.c:1345`.

ok, done

**Review by JaylinYu — COMMENTED** [review]
how about the old conf

**Pull request merged**